connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -1215,17 +1195,13 @@ public void testSuppressCloseErrors() {
 
         workerTask.initialize(TASK_CONFIG);
         workerTask.initializeAndStart();
-        try {
-            workerTask.execute();
-            fail("workerTask.execute should have thrown an exception");
-        } catch (ConnectException e) {
-            assertSame("Exception from put should be the cause", putException, e.getCause());
-            assertTrue("Exception from close should be suppressed", e.getSuppressed().length > 0);
-            assertSame(closeException, e.getSuppressed()[0]);
-        }
+
+        Throwable thrownException = assertThrows(ConnectException.class, () -> workerTask.execute());
+        assertSame("Exception from put should be the cause", putException, thrownException.getCause());
```

Comment:
While `assertSame` and `assertEquals` act the same when the default equals implementation is provided (which compare references)  but I'm curious why are we opting in for `assertSame` over `assertEquals`. Do we need the test to explicitly comparing object reference or just value?

Reply:
This is a remnant of the original implementation. As you correctly guessed, there is no difference in this case between using `assertEquals` and `assertSame`: neither the `RuntimeException` class nor any of its parents override the `equals()` method, so they both are effectively checking for reference equality (as implemented by the base `Object` class)

I'm happy to change these to `assertEquals` (this is the only place where this is used)

Reply:
For me it is more about the readability. When I see `assertSame` it makes me think we are comparing object ref while `assertEquals` for me means we just care about executing `equal` method. I personally would lean toward using `assertEquals`

Reply:
When I wrote this test originally, I did intentionally use assertSame to do a reference equality check.

Reply:
I can't see why we care about the ref in some of these tests but if this is the intent then keeping `assertSame` is okay. My comment was more a question whether we need the test to be explicit or not.

Reply:
@OmniaGM @gharris1727 I'd like to ship it tomorrow if both of you have no objection.